### ISSUE TITLE
Com Exception while getting help file. Reflection API

### ISSUE THREAD

**Issue by joshuader6** [rubberduck-vba/Rubberduck#1538]
I was just playing around with the Reflection API and tested this: 

While hitting the duck.Parse command this happend
![image](https://cloud.githubusercontent.com/assets/10895412/15389672/5a78bd20-1db8-11e6-92a5-981943f1f954.png)
```

 public void SetModuleState(VBComponent component, ParserState state, SyntaxErrorException parserError = null)
        {
            if (AllUserDeclarations.Any())
            {
                var projectId = component.Collection.Parent.HelpFile;
                var project = AllUserDeclarations.SingleOrDefault(item =>
                    item.DeclarationType == DeclarationType.Project && item.ProjectId == projectId);

                if (project == null)
                {
                    // ghost component shouldn't even exist
                    ClearStateCache(component);
                    Status = EvaluateParserState();
                    return;
                }
            }
            var key = new QualifiedModuleName(component);

            _moduleStates.AddOrUpdate(key, new ModuleState(state), (c, e) => e.SetState(state));
            _moduleStates.AddOrUpdate(key, new ModuleState(parserError), (c, e) => e.SetModuleException(parserError));
            Debug.WriteLine("Module '{0}' state is changing to '{1}' (thread {2})", key.ComponentName, state, Thread.CurrentThread.ManagedThreadId);
            OnModuleStateChanged(component, state);
            Status = EvaluateParserState();
        }
```

I cannot seem to get the local information on Component: 
![image](https://cloud.githubusercontent.com/assets/10895412/15389800/f69c28cc-1db8-11e6-84b1-d5e0e7cec45d.png)

Pressing F5 gets me this: 
![image](https://cloud.githubusercontent.com/assets/10895412/15389824/159b0f5e-1db9-11e6-8fe0-b95411510649.png)

```
        private void ParseComponent(VBComponent component, TokenStreamRewriter rewriter = null)
        {
            ParseAsync(component, CancellationToken.None, rewriter).Wait();
        }
```
In this step the Component doesn't throw an error.
![image](https://cloud.githubusercontent.com/assets/10895412/15389884/58384b92-1db9-11e6-81bf-f1118e943e18.png)

Noted that "TestDeclarations" (Component.Name) is the module where my reflection code sits.

This resulted then in this: 
![image](https://cloud.githubusercontent.com/assets/10895412/15389710/7ebca534-1db8-11e6-8162-96a1992e3ebd.png)
```
Option Explicit


Sub tester()
    Dim duck As New Rubberduck.ParserState
    'Set duck = CreateObject("Rubberduck.ParserState")
    duck.Initialize Application.VBE
    Debug.Print "Parsing"
    duck.Parse
    
    Dim i As Variant
    Dim item As Declaration
    Stop
    For Each i In duck.UserDeclarations
        Set item = i
        If item.DeclarationType = DeclarationType_ClassModule Then
            Debug.Print item.Name
        End If
    Next
    
End Sub
```

Strange is that the main parser in the addin works fine without any error.

**Comment by retailcoder** [comment]
Can't reproduce.... what host are you in?

**Comment by joshuader6** [comment]
Excel 
This is in my workbook with very much code 
normal parser works fine but this happens when using the api

**Comment by joshuader6** [comment]
These are the last few lines of the immediate window: 

```
Module 'ISliderEvents' was modified
Module 'ThreeDFormat' was modified
Module 'ISAXErrorHandler' was modified
Module 'ILicAgent' was modified
Module 'TableNames' was NOT modified
Exception thrown: 'System.Runtime.InteropServices.COMException' in Rubberduck.VBEditor.dll
Exception thrown: 'System.Runtime.InteropServices.COMException' in Rubberduck.Parsing.dll
Exception thrown in thread 20:
System.Runtime.InteropServices.COMException (0x80010100): Systemaufruf ist fehlgeschlagen. (Ausnahme von HRESULT: 0x80010100 (RPC_E_SYS_CALL_FAILED))
   bei Microsoft.Vbe.Interop._VBComponent.get_Name()
   bei Rubberduck.Parsing.VBA.RubberduckParser.ResolveDeclarations(VBComponent component, IParseTree tree) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParser.cs:Zeile 516.
   bei Rubberduck.Parsing.VBA.RubberduckParser.ResolveInternal(CancellationToken token) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParser.cs:Zeile 454.
   bei Rubberduck.Parsing.VBA.RubberduckParser.Resolve(CancellationToken token) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParser.cs:Zeile 433.
   bei Rubberduck.Parsing.VBA.RubberduckParser.StateOnStateChanged(Object sender, EventArgs e) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParser.cs:Zeile 68.
   bei System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   bei Rubberduck.Parsing.VBA.RubberduckParserState.OnStateChanged(ParserState state) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParserState.cs:Zeile 139.
   bei Rubberduck.Parsing.VBA.RubberduckParserState.set_Status(ParserState value) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParserState.cs:Zeile 289.
   bei Rubberduck.Parsing.VBA.RubberduckParserState.SetModuleState(VBComponent component, ParserState state, SyntaxErrorException parserError) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParserState.cs:Zeile 189.
   bei Rubberduck.Parsing.VBA.RubberduckParser.<>c__DisplayClass23_0.<ParseAsyncInternal>b__1(Object sender, ParseCompletionArgs e) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\RubberduckParser.cs:Zeile 412.
   bei Rubberduck.Parsing.VBA.ComponentParseTask.Start(CancellationToken token) in C:\Users\jk22291\Documents\Rubberduck\Rubberduck.Parsing\VBA\ComponentParseTask.cs:Zeile 69.
Exception thrown: 'System.Runtime.InteropServices.COMException' in Rubberduck.Parsing.dll
```

**Comment by Vogel612** [comment]
>   bei Rubberduck.Parsing.VBA.RubberduckParser.ResolveDeclarations

This suggests that the problem is in fact the resolver and not the Parser. The difference is a little ... unclear, especially since the class that ties the Parsing and Resolving together is called `RubberduckParser`... Maybe that needs a rename :confused: 

Do you by any chance get a ResolverError when refreshing Rubberduck through the status bar?

**Comment by joshuader6** [comment]
@Vogel612 nope thats the strange thing .. this thing is working fine. Except for the with clause problem #1539 
But this doesn't cause Parser errors

**Comment by joshuader6** [comment]
Seems fixed as of 83b08c4500fd13f945b6589934ae2fa2381f9ff8
